swap/swap.go

```diff
@@ -373,38 +345,41 @@ func (s *Swap) loadLastSentCheque(peer enode.ID) (err error) {
 	//last cheque in memory
 	var cheque *Cheque
 	if _, ok := s.cheques[peer]; !ok {
-		err = s.stateStore.Get(sentChequeKey(peer), &cheque)
-		s.cheques[peer] = cheque
+		err = s.store.Get(sentChequeKey(peer), &cheque)
+		if err != nil && err != state.ErrNotFound {
+			s.cheques[peer] = cheque
+		}
 	}
-	return
+	return err
 }
 
-// saveLastReceivedCheque loads the last received cheque for peer
+// loadLastReceivedCheque loads the last received cheque for peer
 func (s *Swap) loadLastReceivedCheque(peer enode.ID) (cheque *Cheque) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
-	s.stateStore.Get(receivedChequeKey(peer), &cheque)
+	s.store.Get(receivedChequeKey(peer), &cheque)
 	return
 }
 
 // saveLastReceivedCheque saves cheque as the last received cheque for peer
 func (s *Swap) saveLastReceivedCheque(peer enode.ID, cheque *Cheque) error {
 	s.lock.Lock()
 	defer s.lock.Unlock()
-	return s.stateStore.Put(receivedChequeKey(peer), cheque)
+	return s.store.Put(receivedChequeKey(peer), cheque)
 }
 
 // Close cleans up swap
 func (s *Swap) Close() {
-	s.stateStore.Close()
+	s.store.Close()
 }
 
 // resetBalance is called:
 // * for the creditor: on cheque receival
 // * for the debitor: on confirmation receival
-func (s *Swap) resetBalance(peerID enode.ID, amount int64) {
-	log.Info("resetting balance for peer", "peer", peerID.String(), "amount", amount)
-	s.updateBalance(peerID, amount)
+func (s *Swap) resetBalance(peerID enode.ID, amount int64) error {
```

Comment:
in the comments: receival is not a word: upon receiving the cheque

Reply:
Fixed, thanks.